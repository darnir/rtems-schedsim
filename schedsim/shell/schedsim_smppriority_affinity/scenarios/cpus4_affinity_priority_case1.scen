#
#  Ensure lowest priority is preempted
#
rtems_init 4
task_create smp1 7
task_create smp2 8
task_create smp3 5
task_create smp4 6
task_create smp5 9

# GOAL: want 0x0a010001 - 0x0a010004 as heir and executings
# GOAL:   on CPUs 3-0 respectively
cpus smp4 smp3 smp2 smp1

# Tasks are created and distributed across CPUs
task_set_affinity smp5 0x01
# GOAL: want 0x0a010001 - 0x0a010004 as heir and executings
# GOAL:   on CPUs 3-0 respectively
# PRI: 6    5    8   7
cpus smp4 smp3 smp2 smp1

task_priority smp5 5
# GOAL: want 0x0a010005, 0x0a010003, 0x0a010004, 0x0a010001
# GOAL:   to be heir AND executing on CPU 0-3 respectively
# PRI: 5    5    6   7
cpus smp5 smp3 smp4 smp1
