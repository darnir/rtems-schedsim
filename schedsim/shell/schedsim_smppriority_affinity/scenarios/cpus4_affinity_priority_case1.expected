Script File               : scenarios/cpus4_affinity_priority_case1.scen
verbose                   : 1
==> 1: #
==> 2: #  Ensure lowest priority is preempted
==> 3: #
==> 4: rtems_init 4
  CPU 0: Thread Heir 0x09010001 priority=255
  CPU 0: Thread Executing 0x09010001 priority=255
  CPU 1: Thread Heir 0x09010002 priority=255
  CPU 1: Thread Executing 0x09010002 priority=255
  CPU 2: Thread Heir 0x09010003 priority=255
  CPU 2: Thread Executing 0x09010003 priority=255
  CPU 3: Thread Heir 0x09010004 priority=255
  CPU 3: Thread Executing 0x09010004 priority=255
==> 5: task_create smp1 7
Creating task with:  Preemption: yes Timeslicing: no
Task (smp1) created: id=0x0a010001, priority=7
Task (smp1) starting: id=0x0a010001, priority=7
  CPU 3: Thread Heir 0x0a010001 priority=7
  CPU 3: Thread Executing 0x0a010001 priority=7
==> 6: task_create smp2 8
Creating task with:  Preemption: yes Timeslicing: no
Task (smp2) created: id=0x0a010002, priority=8
Task (smp2) starting: id=0x0a010002, priority=8
  CPU 2: Thread Heir 0x0a010002 priority=8
  CPU 2: Thread Executing 0x0a010002 priority=8
==> 7: task_create smp3 5
Creating task with:  Preemption: yes Timeslicing: no
Task (smp3) created: id=0x0a010003, priority=5
Task (smp3) starting: id=0x0a010003, priority=5
  CPU 1: Thread Heir 0x0a010003 priority=5
  CPU 1: Thread Executing 0x0a010003 priority=5
==> 8: task_create smp4 6
Creating task with:  Preemption: yes Timeslicing: no
Task (smp4) created: id=0x0a010004, priority=6
Task (smp4) starting: id=0x0a010004, priority=6
  CPU 0: Thread Heir 0x0a010004 priority=6
  CPU 0: Thread Executing 0x0a010004 priority=6
==> 9: task_create smp5 9
Creating task with:  Preemption: yes Timeslicing: no
Task (smp5) created: id=0x0a010005, priority=9
Task (smp5) starting: id=0x0a010005, priority=9
==> 10: 
==> 11: # GOAL: want 0x0a010001 - 0x0a010004 as heir and executings
==> 12: # GOAL:   on CPUs 3-0 respectively
==> 13: cpus smp4 smp3 smp2 smp1
=== CPU Status
          EXECUTING      /   HEIR             / SWITCH NEEDED
  CPU 0: 0x0a010004 @  6 / 0x0a010004 @  6          false
  CPU 1: 0x0a010003 @  5 / 0x0a010003 @  5          false
  CPU 2: 0x0a010002 @  8 / 0x0a010002 @  8          false
  CPU 3: 0x0a010001 @  7 / 0x0a010001 @  7          false
=== End of Ready Set of Threads
==> 14: 
==> 15: # Tasks are created and distributed across CPUs
==> 16: task_set_affinity smp5 0x01
Task (0x0a010005) Set affinity=0x00000001
==> 17: # GOAL: want 0x0a010001 - 0x0a010004 as heir and executings
==> 18: # GOAL:   on CPUs 3-0 respectively
==> 19: # PRI: 6    5    8   7
==> 20: cpus smp4 smp3 smp2 smp1
=== CPU Status
          EXECUTING      /   HEIR             / SWITCH NEEDED
  CPU 0: 0x0a010004 @  6 / 0x0a010004 @  6          false
  CPU 1: 0x0a010003 @  5 / 0x0a010003 @  5          false
  CPU 2: 0x0a010002 @  8 / 0x0a010002 @  8          false
  CPU 3: 0x0a010001 @  7 / 0x0a010001 @  7          false
=== End of Ready Set of Threads
==> 21: 
==> 22: task_priority smp5 5
  CPU 0: Thread Heir 0x0a010005 priority=5
  CPU 0: Thread Executing 0x0a010005 priority=5
  CPU 2: Thread Heir 0x0a010004 priority=6
  CPU 2: Thread Executing 0x0a010004 priority=6
Task (0x0a010005) Change Priority from 9 to 5
==> 23: # GOAL: want 0x0a010005, 0x0a010003, 0x0a010004, 0x0a010001
==> 24: # GOAL:   to be heir AND executing on CPU 0-3 respectively
==> 25: # PRI: 5    5    6   7
==> 26: cpus smp5 smp3 smp4 smp1
=== CPU Status
          EXECUTING      /   HEIR             / SWITCH NEEDED
  CPU 0: 0x0a010005 @  5 / 0x0a010005 @  5          false
  CPU 1: 0x0a010003 @  5 / 0x0a010003 @  5          false
  CPU 2: 0x0a010004 @  6 / 0x0a010004 @  6          false
  CPU 3: 0x0a010001 @  7 / 0x0a010001 @  7          false
=== End of Ready Set of Threads
